pkg/api/transport_errcatch.go

```diff
@@ -55,9 +62,29 @@ func (e *ErrCatchTransport) RoundTrip(req *http.Request) (*http.Response, error)
 
 	res, err := e.rt.RoundTrip(req)
 	if res != nil {
-		// nolint
-		httputil.DumpResponse(res, res.Body != nil)
+		_, _ = httputil.DumpResponse(res, res.Body != nil)
+
+		// When the content type is "text/html", a bit of tweaking is required
+		// for the response to be marshaled to  JSON. Using the standard error
+		// definition and populating it with parts of the request so the error
+		// can be identified.
+		if strings.Contains(res.Header.Get("Content-Type"), "text/html") {
```

Comment:
I assume we don't expect the `Content-type` header value to be something else - most probably my mistake.... ?

Reply:
Also it would make sense to make those content-type as constants and I wonder if they already exist in some core networking package.

Reply:
As it turns out, it does have more info, it contains encoding information, that's why I used `strings.Contains()`, this is the header value:

```
Content-Type: text/html; charset=UTF-8
```